.travis/zip_assets

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -eu
+(cd game_headed_assets || exit 1; zip -r game_headed_assets.zip -- *)
```

Comment:
Why does this line look so complicated?

Reply:
Combination of 'zip' and 'shellcheck'.

'zip' apparently does not have a flag for the 'base path' of the zip file. Zipping files from the current location creates a zip with paths "game_headed_assets/...". The zip files are extracted into "assets" and should not include the "game_headed_assets" leading folder. So there is a 'cd' command so that the zip contents is just the set of folders in 'game_headed_assets'.

With the 'cd', comes the sub-shell, the parens, so that we exit back out to the original folder once the command is done.

Shellcheck then flagged a couple of things. It said there should be a `cd || exit` in case the cd command fails. It also said that the glob on the zip task should be done with '--' to avoid issues with file names containing dashes and symbols in them.

Reply:
Ok, makes sense.
Feel free to merge, but considering clarifying this a bit. I knew that the zip format stores relative paths (also absolute, but those are commonly blocked for security reasons), but the first time I saw the command I'd just have assumed I could move all the individual parts of the command into a single line

Reply:
A lot of the documentation on this script is very generic. Hard to think what it should be without copy/pasting the first response I had into the script, and that seems to be too low level and perhaps duplicative.

I created a wiki page for 'shell script conventions' to perhaps better document this: https://github.com/triplea-game/triplea/wiki/Shell-Script-Conventions